tf_encrypted/protocol/pond.py

```diff
@@ -3377,31 +3432,46 @@ def _mask_private(prot: Pond, x: PondPrivateTensor) -> PondMaskedTensor:
     x0, x1 = x.unwrapped
 
     with tf.name_scope("mask"):
+        with tf.device(prot.crypto_producer.device_name):
+            s0 = seed()
+            s1 = seed()
+            a0seed = x.backing_dtype.seeded_tensor(x.shape, s0)
+            a1seed = x.backing_dtype.seeded_tensor(x.shape, s1)
+
+        return PondMaskedTensor(prot, x, None, a0seed, a1seed, None, None, x.is_scaled)
```

Comment:
this worries me :D don't like optional values if we can avoid it

Reply:
keep in mind that we're building a graph, so you're defining computations, not running them